wfcGen.py

```diff
@@ -0,0 +1,244 @@
+import random
+
+from PIL import Image
+
+
+res = 3
+cropShift = False
+cells = {}
+cropped = []
+for ins in range(4):
+    field = Image.open("image.png")
+    cropped.append([])
+
+    if cropShift:
+        field = field.rotate(90*ins)
+
+        SX, SY = field.size
+
+        copy = Image.new("RGB", (SX*2, SY*2))
+        copy.paste(field, (0, 0))
+        copy.paste(field, (SX, 0))
+        copy.paste(field, (0, SY))
+        copy.paste(field, (SX, SY))
+        field = copy.copy()
+
+        cropped[ins] = [[field.crop((x, y, x+res, y+res)) for x in range(SX)] for y in range(SY)]
+
+    else:
+        field = field.rotate(90*ins)
+        SX = field.size[0]//res*res
+        SY = field.size[1]//res*res
+        field = field.crop((0, 0, SX, SY))
+
+        cropped[ins] = [[field.crop((x*res, y*res, x*res+res, y*res+res)) for x in range(SX//res)] for y in range(SY//res)]
+
+    lcy = len(cropped[ins])
+    lcx = len(cropped[ins][0])
+
+    print(lcy, lcx)
+
+
+
+    # [left, top, right, bottom]uwu
+    # u will newer need that
+
+    for y in range(lcy):
+        for x in range(lcx):
+            flag = -1
+            for key in cells.keys():
+                if cropped[ins][y][x] == cells[key]["img"]:
+                    flag = key
+
+            if flag != -1:
+                cells[flag]["nbg"]["0"].append(cropped[ins][y][x-1-(res-1)*int(cropShift)])
+                cells[flag]["nbg"]["1"].append(cropped[ins][y-1-(res-1)*int(cropShift)][x])
+                cells[flag]["nbg"]["2"].append(cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx])
+                cells[flag]["nbg"]["3"].append(cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x])
+            else:
+                cells[str(len(cells.keys()))] = {
+                "nbg" : {
+                    "0" : [cropped[ins][y][x-1-(res-1)*int(cropShift)]],
+                    "1" : [cropped[ins][y-1-(res-1)*int(cropShift)][x]],
+                    "2" : [cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx]],
+                    "3" : [cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x]],
+                }, 
+                "img" : cropped[ins][y][x]}

```

Comment:
Formatting is not very easy to read here:
```suggestion
                cells[str(len(cells.keys()))] = {
                    "nbg" : {
                        "0" : [cropped[ins][y][x-1-(res-1)*int(cropShift)]],
                        "1" : [cropped[ins][y-1-(res-1)*int(cropShift)][x]],
                        "2" : [cropped[ins][y][(x+1+(res-1)*int(cropShift))%lcx]],
                        "3" : [cropped[ins][(y+1+(res-1)*int(cropShift))%lcy][x]],
                    }, 
                    "img" : cropped[ins][y][x]
                }
```